### PR TITLE
Feature/add icons

### PR DESCRIPTION
**Pull request by iaronaraujo**
#### What is the purpose of this pull request?
Added the `iconProps` prop to `MenuItem`, making it possible to add an icon to the left or to the right of the item's text.

#### What problem is this solving?
https://github.com/vtex-apps/store-discussion/issues/4

#### How should this be manually tested?
[workspace](https://iaronaraujo--storecomponents.myvtex.com/)

#### Screenshots or example usage
image

#### Types of changes
- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [x] Requires change to documentation, which has been updated accordingly.

### CODE REVIEW

**Comment by jgfidelis** [comment]
lgtm but I want to wait for the opinion of a second person before approving

File: README.md [hunk 1]
```diff
@@ -41,23 +41,39 @@ To configure or customize this app, you need to import it in your dependencies i
...
+Now, you can change the behavior of the menu block that is in the store. See an example of how to configure:
 
 ```json
-"menu-link": {
+"vtex.menu@2.x:menu#websites": {
+    "children": [
+      "menu-item#shop",
+      "menu-item#about-us"
+    ]
+  },
```
Comment (klzns) [inline README.md]: Fix indent

File: README.md [hunk 2]
```diff
@@ -41,23 +41,39 @@ To configure or customize this app, you need to import it in your dependencies i
...
+        "text": "Shop"
+      },
+      "iconProps": {
+        "id": "bnd-logo",
+        "size": 16,
+        "viewBox": "0 0 16 16",
+        "activeClassName": "rebel-pink",
+        "mutedClassName": "c-action-primary"
+      },
+      "iconToTheRight": true
```
Comment (klzns) [inline README.md]: Same

File: README.md [hunk 3]
```diff
@@ -74,23 +90,46 @@ The menu has no required or allowed block. So, any menu block implementation do
...
 
-Through the Storefront, you can change the menu's behavior and interface. However, you also can make in your theme app, as Store theme does.
+Through the Storefront, you can change the menu's behavior and interface. However, you also can make in your theme app, as Store theme does. The menu have `menu-item` as children. The props of the `menu-item` are as it follows:
+
+| Prop name      | Type     | Description                                          | Default value |
+| -------------- | -------- | ---------------------------------------------------- | ------------- |
+| `type`         | `String` | the type of the menu item. Can be `category` or `custom`                                            | `caterogy`           |
+| `id`         | `String` | the ID of the menu item                                            | N/A           |
+| `highlight`         | `boolean` | if the item should have highlight                                            | N/A           |
+| `iconToTheRight`         | `boolean` | if the icon should be on the right or on the left of the menu text                                           | N/A           |
```
Comment (klzns) [inline README.md]: Default value is `false`

File: react/__tests__/StyledLink.test.tsx [hunk 4]
```diff
@@ -0,0 +1,66 @@
+import { render } from '@vtex/test-tools/react'
+import React from 'react'
+
+import StyledLink from '../components/StyledLink'
+
+
+describe('Icon should appear at the left', () => {
+    function renderComponent(toTheRight: boolean) {
+        const props = {
```
Comment (klzns) [inline react/__tests__/StyledLink.test.tsx]: Indent with 2 spaces instead of 4

File: react/components/StyledLink.tsx [hunk 5]
```diff
@@ -42,9 +44,24 @@ const StyledLink: FunctionComponent<StyledLinkProps> = props => {
     pointer: !disabled,
   })
 
+  const iconTestId = `icon-${iconToTheRight ? 'right' : 'left'}`
+  const iconComponent = iconProps ?
+    <span className="mh2" data-testid={iconTestId}>
```
Comment (klzns) [inline react/components/StyledLink.tsx]: Add a CSS handle here

File: react/typings/vtex.store-icons.d.ts [hunk 6]
```diff
@@ -0,0 +1,3 @@
+declare module 'vtex.store-icons' {
+    export const Icon
```
Comment (klzns) [inline react/typings/vtex.store-icons.d.ts]: Fix indent

File: yarn.lock [hunk 7]
```diff
@@ -0,0 +1,4 @@
+# THIS IS AN AUTOGENERATED FILE. DO NOT EDIT THIS FILE DIRECTLY.
```
Comment (klzns) [inline yarn.lock]: Remove file

File: react/MenuItem.tsx [hunk 8]
```diff
@@ -57,7 +58,8 @@ const MenuItem: StorefrontFunctionComponent<MenuItemSchema> = ({
 export interface MenuItemSchema {
   id: string
   type: 'category' | 'custom'
-  iconId: string
+  iconProps: IconProps
+  iconToTheRight: boolean
```
Comment (lbebber) [inline react/MenuItem.tsx]: acho que seria melhor uma prop `iconPosition` que aceita `left` ou `right`

**Pull request merged**